### PR TITLE
Create ckks_mul.hpp

### PR DESCRIPTION
**Pull request by bbqgonewrong**
added the multiplier function for ckks

### CODE REVIEW

File: BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    auto encryptedArray = context.ea->getCx();
+    
+    
+    //Ctxt c1(pubKey), c2(pubKey);
+    // Encrypt our float values as complex number into our ciphertext
+    vector<complex<double>> vd1 = { complex<double>(a) };
+    vector<complex<double>> vd2 = { complex<double>(b) };
+
+    
+  helib::Ctxt c1(pubKey), c2(pubKey);
```
Comment (HindrikStegenga) [inline BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp]: Fix your indentation will you?

File: BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  //Encrypt the values
+  encryptedArray.encrypt(c1, pubKey, vd1);
+  encryptedArray.encrypt(c2, pubKey, vd2);
+  NTL::xdouble expectedPtxtMag = c1.getPtxtMag() * c2.getPtxtMag();
+  c1 *= c2;
+  
+  //Decrypting the values
+  encryptedArray.decrypt(c1, secretKey, vd3);
+  vector<complex<double>> actual = {};
+
```
Comment (HindrikStegenga) [inline BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp:63]: What are you doing here???
The only change between my code and yours should be a += to a *=....

File: BGV-Adder/Algorithms/CKKSAdd/ckks_mul.hpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+#ifndef ckks_mul_hpp
+#define ckks_mul_hpp
+
+#include <stdio.h>
+#include <helib/helib.h>
+
+double calcMaxDiff(const std::vector<std::complex<double>>& v1,
+                   const std::vector<std::complex<double>>& v2);
+
+double ckks_add_doubles(double a, double b);
```
Comment (HindrikStegenga) [inline BGV-Adder/Algorithms/CKKSAdd/ckks_mul.hpp]: Please fix the function name and remove the calcMaxDiff one.

**Review by HindrikStegenga — CHANGES_REQUESTED** [review]

File: BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp (hunk 1 above)
Comment (bbqgonewrong) [reply BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp]: ok

File: BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp (hunk 2 above)
Comment (bbqgonewrong) [reply BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp:63]: the actual vector stores the actual multiplication as it is not a straight forward a*b for me

Comment (bbqgonewrong) [reply BGV-Adder/Algorithms/CKKSAdd/ckks_mul.cpp:63]: i mean the non-encrypted one

**Pull request merged**